### PR TITLE
Support nondeterministic collations for LIKE in CHECK constraint clause

### PR DESCRIPTION
**Pull request by AvantikaDG**
Utility statements such as ALTER TABLE are not planned. Due to collation like to ilike transformation logic being called in a planner hook, CHECK constraint in utlity commands do not support LIKE for nondeterministic collations.

To solve this problem, we add a hook to invoke the like to ilike transformation function after the raw parsetree is transformed into an executable expression.

Task: BABEL-3319
Signed-off-by: Avantika Dasgupta <davantik@amazon.com>

Extension PR: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/684

### CODE REVIEW

File: src/backend/catalog/heap.c [hunk 1]
```diff
@@ -3225,6 +3227,19 @@ cookConstraint(ParseState *pstate,
...
+	/* 
+	 * If there is a LIKE operation within CHECK constraint, then call the hook to 
+	 * transform like node into ilike node for tsql compatibility. Since utility 
+	 * statements do not get planned, we invoke a planner hook here to carry out the
+	 * transformation. 
+	 * Note that we cannot guarantee that this hook has not already
+	 * been invoked, but the invoked function must take care of preventing any
+	 * duplicated transformations.
+	 */
+	if(IsA(raw_constraint, A_Expr) && ((A_Expr*)raw_constraint)->kind == AEXPR_LIKE
```
Comment (Deepesh125) [inline src/backend/catalog/heap.c]: I think, It would be better to have dialect check as well.

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

**Comment by 2jungkook** [comment]
Please add an extension PR link to the description (same for the extension PR description).

File: src/backend/catalog/heap.c [hunk 2]
```diff
@@ -3225,6 +3227,18 @@ cookConstraint(ParseState *pstate,
 	 */
 	assign_expr_collations(pstate, expr);
 
+	/* 
+	 * If there is a LIKE operation within CHECK constraint, then call the hook to 
+	 * transform like node into ilike node for tsql compatibility. Since utility 
+	 * statements do not get planned, we invoke a planner hook here to carry out the
```
Comment (2jungkook) [inline src/backend/catalog/heap.c]: planner hook?

File: src/backend/catalog/heap.c [hunk 3]
```diff
@@ -68,6 +68,7 @@
 #include "executor/spi.h"
 #include "miscadmin.h"
 #include "nodes/nodeFuncs.h"
+#include "nodes/nodes.h"
```
Comment (2jungkook) [inline src/backend/catalog/heap.c]: why do we need this?

**Review by 2jungkook — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**